File: lib/rgfa/line/group/path/captured_path.rb
module RGFA::Line::Group::Path::CapturedPath

  def captured_edges
    if !connected?
      raise RGFA::RuntimeError,
        "Captured path cannot be computed\n"+
        "Line is not connected to a RGFA instance\n"+
        "Line: #{self}"
    end
    links
  end

  def captured_segments
    if !connected?
      raise RGFA::RuntimeError,
        "Captured path cannot be computed\n"+
        "Line is not connected to a RGFA instance\n"+
        "Line: #{self}"
    end
    segment_names
  end

  def captured_path
    if !connected?
      raise RGFA::RuntimeError,
        "Captured path cannot be computed\n"+
        "Line is not connected to a RGFA instance\n"+
        "Line: #{self}"
    end
    retval = []
    (segment_names.size-1).times do |i|
      retval << segment_names[i]
      retval << links[i]
    end
    retval << segment_names[-1]
    if segment_names.size == links.size
      retval << links[-1]
      retval << segment_names[0]
    end
    return retval
  end

end

File: gfapy/line/group/path/captured_path.py
import gfapy

class CapturedPath:

  @property
  def captured_edges(self):
    if not self.is_connected():
      raise gfapy.RuntimeError(
        "Captured path cannot be computed\n"+
        "Line is not connected to a GFA instance\n"+
        "Line: {}".format(self))
    return self.links

  @property
  def captured_segments(self):
    if not self.is_connected():
      raise gfapy.RuntimeError(
        "Captured path cannot be computed\n"+
        "Line is not connected to a GFA instance\n"+
        "Line: {}".format(self))
    return self.segment_names

  @property
  def captured_path(self):
    if not self.is_connected():
      raise gfapy.RuntimeError(
        "Captured path cannot be computed\n"+
        "Line is not connected to a GFA instance\n"+
        "Line: {}".format(self))
    retval = []
    if len(self.segment_names) == 1:
      retval.append(self.segment_names[0])
    else:
      for i in range(len(self.segment_names) - 1):
        retval.append(self.segment_names[i])
        retval.append(self.links[i])
      retval.append(self.segment_names[-1])
      if len(self.segment_names) == len(self.links):
        retval.append(self.links[-1])
        retval.append(self.segment_names[0])
    return retval
